Fixes #15793 - quick provisioning sets build mode

lzap opened the pull request:
Regression introduced by our modal dialog.

Needed for successful testing of

https://github.com/theforeman/foreman_discovery/pull/226

lzap committed:
Fixes #15793 - quick provisioning sets build mode

lzap commented:
Found one additional issue, hold on.

dLobatog commented on app/controllers/discovered_hosts_controller.rb:
```diff
@@ -71,6 +71,7 @@ def update
 
   def perform_update(host, success_message = nil)
     Taxonomy.no_taxonomy_scope do
+      host.build = true
```
Please add a test that ensures `host.build` is true when we do this action or something, that way we won't get this regression again

lzap replied:
Will do

lzap commented:
So the problem was bigger. First, quick provisioning did not set the build flag, this was easy. But the flag can't be set just like `host.build = true`, we need to execute some statements from `to_managed` in ordere to setup things (legacy flag, fact clear override). Therefore I had to extract it into separate method. Added a unit test for that.

But during testing I figured out that our new discovery_clean_facts setting is implemented incorrectly. We want to let users to pick either "delete none" or "delete only non-discovered facts" while it was implemented with choice of "delete all" or "delete only non-discovered facts". So it was not useful at all.

I fixed that as well.

dLobatog commented:
Tests run fine locally - It's also able to clean up facts & provision fine, so I'm merging

dLobatog commented:
Merged as 454c6fa0833686587a9afd6a9fbeee01c4927efd, thanks @lzap!
